cob_script_server/src/simple_script_server/simple_script_server.py

```diff
@@ -782,6 +785,85 @@ def publish_twist(self, pub, twist, end_time):
 			pub.publish(twist)
 			r.sleep()
 
+	## Relative movement
+	## Deals with all kind of relative movements for different components.
+	#
+	# Based on the component, the corresponding move functions will be called.
+	#
+	# \param component_name Name of the component.
+	# \param parameter_name Name of the parameter on the ROS parameter server.
+	# \param blocking Bool value to specify blocking behaviour.
+	#
+	# # throws error code 3 in case of invalid parameter_name vector
+	def move_rel(self, component_name, parameter_name, blocking=True):
+		ah = action_handle("move_rel", component_name, parameter_name, blocking, self.parse)
+		if(self.parse):
+			return ah
+		else:
+			ah.set_active(mode="topic")
+
+		rospy.loginfo("Move <<%s>> relatively by <<%s>>", component_name, parameter_name)
+
+		# step 0: check validity of parameters:
+		if not isinstance(parameter_name, list):
+			message = "Parameter " + str(parameter_name) + " not formated correctly (not a list of numeric lists), aborting move_rel"
+			rospy.logerr(message)
+			print("parameter_name must be list")
+			ah.set_failed(3, message)
+			return ah
+		for parameter in parameter_name:
+			if not isinstance(parameter, list):
+				message = "Parameter " + str(parameter_name) + " not formated correctly (not a list of numeric lists), aborting move_rel"
+				rospy.logerr(message)
+				print("parameter_name must be list of lists")
+				ah.set_failed(3, message)
+				return ah
+			for param in parameter:
+				if not isinstance(param, (int, float)):
+					message = "Parameter " + str(parameter) + " not formated correctly (not a list of numeric lists), aborting move_rel"
+					rospy.logerr(message)
+					print("parameter_name must be list of numeric lists")
+					ah.set_failed(3, message)
+					return ah
+
+		# step 1: get current position
+		timeout = 1.0
+		try:
+			joint_names = list(rospy.wait_for_message("/" + component_name + "/joint_states", JointState, timeout = timeout).name)
+			start_pos = list(rospy.wait_for_message("/" + component_name + "/joint_states", JointState, timeout = timeout).position)
+		except rospy.ROSException as e:
+			rospy.logwarn("no joint states received from %s within timeout of %ssec. using default point time of 8sec.", component_name, str(timeout))
```

Comment:
in case we cannot receive joint states, we should abort (`ah.set_failed` and `return ah`)